lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -1343,14 +1372,56 @@ struct LowerTypesPass : public LowerFIRRTLTypesBase<LowerTypesPass> {
 // This is the main entrypoint for the lowering pass.
 void LowerTypesPass::runOnOperation() {
   std::vector<Operation *> ops;
-  llvm::for_each(getOperation().getBody()->getOperations(),
-                 [&](Operation &op) { ops.push_back(&op); });
+  // Record of each module, to the map of NonLocalAnchors which need to be
+  // updated with the lowered inner_sym name.
+  DenseMap<Operation *, DenseMap<StringAttr, StringAttr>>
```

Comment:
Per-thread state you merge or a vector with a lock you collect these into would be a lot cleaner.

Reply:
Used the reduce pattern with a smallvector